Fix/issue-621: Enhance SearchTeamMemberHandler with additional includes

KobMiles opened the pull request:
dev
## JIRA

* [621](https://github.com/ita-social-projects/VictoryCenter-Back/issues/621)

## Summary of issue

Handler for searching team members, returns team members without Image

## Summary of change

Added Incude so that Image is in the team member

## Testing approach

- Go to Swagger
- Call up the team members search
- Check whether there is an Image field in the objects 

## CHECK LIST
- [x]  СI passed
- [x]  Сode coverage >=95%
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Admin team member search and listings now reliably include member images so pictures display correctly.

* **Refactor**
  * Data retrieval adjusted to explicitly include related image and category information without changing external behavior.

* **Tests**
  * Added unit tests and test helpers to cover members with and without images and to validate DTO mapping.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

KobMiles committed:
Enhance SearchTeamMemberHandler with additional includes

Modified the `GetAllAsync` method in `SearchTeamMemberHandler` to eagerly load the `Image` property alongside `Category` for `TeamMember` entities. This ensures more comprehensive data retrieval to support new functionality.

KobMiles committed:
Add tests for TeamMember image handling in search

Updated `SearchTeamMemberTests` to include two new test cases:
1. `Handle_WithImage_ShouldReturnImage` verifies that a `TeamMember` with an associated image is correctly mapped and returned.
2. `Handle_WithoutImage_ShouldReturnNullImage` ensures that a `TeamMember` without an image results in a `null` image in the DTO.

Added a `SetupMapper` helper method to streamline mock mapper setup. Expanded `using` directives to include necessary namespaces for the new functionality.

These changes improve test coverage for scenarios involving `TeamMember` entities with and without images.

KobMiles committed:
Refactor SearchTeamMemberHandler query logic

Simplified the query logic in `SearchTeamMemberHandler` by introducing a `QueryOptions<TeamMember>` object for the `GetAllAsync` method. Consolidated `Include` statements into a single chained expression and added a null-forgiving operator (`!`) to the `Image` property. Removed redundant `Include` statements for improved readability and maintainability. No changes were made to the `Filter`, `Offset`, or `Limit` properties.

maxvonlancaster commented on VictoryCenter/VictoryCenter.UnitTests/MediatRHandlersTests/TeamMembers/SearchTeamMemberTests.cs:
```diff
@@ -110,6 +111,112 @@ public async Task Handle_InvalidFullName_ShouldReturnValidationError()
+            new()
+            {
+                Id = 3,
+                FullName = "No Image",
```
Later to replace with Fixture

KobMiles committed:
Refactor test data creation in SearchTeamMemberTests

Centralized test data creation by introducing static helper methods
(`CreateTeamMember`, `CreateTeamMemberDto`, `CreateImage`, and
`CreateImageDto`) to simplify and standardize object initialization.
Replaced inline object creation in tests with calls to these methods,
reducing duplication and improving readability.

Updated test setups for `Handle_WithImage_ShouldReturnImage` and
`Handle_WithoutImage_ShouldReturnNullImage` to use the new helper
methods. Adjusted `SetupMapper` and `SetupRepositoryWrapper` calls
accordingly.

No functional changes were made to the tests; assertions remain
unchanged. This refactor improves maintainability and clarity of the
test code.

KobMiles committed:
Merge branch 'release/1.0.0' into fix/issue-621

KobMiles committed:
Refactor CreateTeamMemberDto to remove email parameter

The `CreateTeamMemberDto` method signature was updated to remove the `email` parameter, simplifying its usage. All calls to this method in the `SearchTeamMemberTests` class were updated accordingly, and the `Email` property assignment was removed from the method implementation.

Test cases were adjusted to align with this change:
- Removed the `email` argument from `CreateTeamMemberDto` calls in tests involving team members with and without images.

This change reflects a design decision to handle `email` differently or exclude it from the `TeamMemberDto` object.

KobMiles committed:
Add ImageId property to object initialization

Introduced a new `ImageId` property in the `SearchTeamMemberTests` class. The property is initialized with `image?.Id` to safely retrieve the `Id` of the `image` object if it is not null. This ensures the `ImageId` is captured alongside other properties like `Priority`, `CategoryId`, `Status`, `Description`, `Email`, and `Image`.
